moma_mission/demo/piloting.py

```diff
@@ -38,19 +39,31 @@
                                                                                        'Failure': 'Failure',
                                                                                        'NextWaypoint': 'WAYPOINT_FOLLOWING'})
 
-        state_machine.add('DETECTION', detection_sequence_factory(), transitions={'Success': 'LATERAL_MANIPULATION',
+        state_machine.add('DETECTION', detection_sequence_factory(), transitions={'Success': 'APPROACH_VALVE',
```

Comment:
Following the manipulation part of the state machine we can fragment the factory in the constituent states. So we have all the logic in a glance.